### PR TITLE
feat(Postgres PGVector Store Node): Add PGVector vector store node

### PR DESCRIPTION
**Pull request by jeanpaul**
## Summary
Add support for the postgres PGVector extension as a vector store. This works similarly to other vector store nodes.

## Related Linear tickets, Github issues, and Community forum posts
https://linear.app/n8n/issue/AI-222/add-support-for-pg-vector-store

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/vector_store/VectorStorePGVector/VectorStorePGVector.node.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+		const pool = pgConf.db.$pool as unknown as pg.Pool;
+
+		const config: PGVectorStoreArgs = {
+			pool,
+			tableName,
+			filter,
+		};
+
+		const collectionOptions = context.getNodeParameter('options.collection.values', 0, {}) as {
+			useCollection?: boolean;
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/nodes/vector_store/VectorStorePGVector/VectorStorePGVector.node.ts:206]: Shall we create a type definition for that?

File: packages/@n8n/nodes-langchain/nodes/vector_store/VectorStorePGVector/VectorStorePGVector.node.ts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+			config.collectionTableName = collectionOptions.collectionTableName;
+		}
+
+		config.columns = context.getNodeParameter('options.columnNames.values', 0, {
+			idColumnName: 'id',
+			vectorColumnName: 'embedding',
+			contentColumnName: 'text',
+			metadataColumnName: 'metadata',
+		}) as {
+			idColumnName: string;
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/nodes/vector_store/VectorStorePGVector/VectorStorePGVector.node.ts:222]: Let's create a type for this as well

**Review by burivuhster — CHANGES_REQUESTED** [review]
Looks good, only two minor notes

**Comment by luizeof** [comment]
Can't wait to use PgVector

**Review by burivuhster — APPROVED** [review]
LGTM! Thanks for addressing my comments

**Review by OlegIvaniv — APPROVED** [review]
Re-approved to trigger E2Es

**Comment by janober** [comment]
Got released with [`n8n@1.58.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.58.0)

**Pull request merged**